library/python-library/src/main/java/uk/gov/gchq/gaffer/python/operation/BuildImageFromDockerfile.java

```diff
@@ -83,20 +84,33 @@ public String buildImage(final String scriptName, final Map<String, Object> scri
         }, buildParam);
     }
 
-    public void getFiles(final String pathAbsolutePythonRepo) throws IOException {
-        String[] fileNames = new String[] {"Dockerfile",
-                                            "DataInputStream.py",
-                                            "entrypoint.py",
-                                            "modules.txt" };
-
-        for (final String fileName : fileNames) {
+    public void getFiles(final String pathAbsolutePythonRepo, final String dockerfilePath) throws IOException {
+        String[] fileNames = new String[] {"DataInputStream.py", "entrypoint.py", "modules.txt"};
+        if (dockerfilePath.equals("")) {
             // Use the default file
-            LOGGER.info("Using the default Dockerfile");
-            InputStream inputStream = StreamUtil.openStream(getClass(), "/.PythonBin/" + fileName);
-            BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
-            String fileData = reader.lines().collect(Collectors.joining(System.lineSeparator()));
-            inputStream.close();
-            Files.write(Paths.get(pathAbsolutePythonRepo + "/" + fileName), fileData.getBytes());
+            LOGGER.info("DockerfilePath unspecified, using default Dockerfile");
+            createFile("Dockerfile", pathAbsolutePythonRepo);
+        } else {
+            LOGGER.info("DockerfilePath specified, using non-default dockerfile");
+            final String[] pathSplit = dockerfilePath.split("/");
+            final String fileName = pathSplit[pathSplit.length - 1];
+            final String fileLocation = dockerfilePath.substring(0, dockerfilePath.length() - fileName.length());
+            createFile(fileName, pathAbsolutePythonRepo, fileLocation);
+        }
+        for (final String fileName : fileNames) {
+            createFile(fileName, pathAbsolutePythonRepo);
         }
     }
+
+    private void createFile(final String fileName, final String destination) throws IOException {
+        createFile(fileName, destination, "/.PythonBin/");
+    }
+
+    private void createFile(final String fileName, final String destination, final String fileLocation) throws IOException {
+        InputStream inputStream = StreamUtil.openStream(getClass(), fileLocation + fileName);
```

Comment:
Consider using a try-with-resources here. Although you're catching the exception at a higher level, the resources will not be disposed of properly if they throw.